waiter/test/waiter/token_test.clj

```diff
@@ -2721,3 +2723,241 @@
                {"maintenance" false "owner" "owner3" "token" "token7"}
                {"maintenance" false "owner" "owner3" "token" "token8"}}
              (set (json/read-str body)))))))
+
+(defmacro assert-channels-no-new-message
+  [chans timeout-ms]
+  `(let [chans# ~chans
+         timeout-ms# ~timeout-ms
+         timeout-chan# (async/timeout timeout-ms#)
+         [_# res-chan#] (-> chans#
+                            (conj timeout-chan#)
+                            (async/alts!! :priority true))]
+     (is (= res-chan# timeout-chan#))))
+
+(defmacro assert-channels-next-message-with-fn
+  [chans msg-fn]
+  `(let [chans# ~chans
+         msg-fn# ~msg-fn
+         res# (for [chan# chans#] (async/<!! chan#))]
+     (is (every? msg-fn# res#))))
+
+(defmacro assert-channels-next-message
+  "Assert that list of channels next message"
+  [chans msg]
+  `(let [chans# ~chans
+         msg# ~msg]
+     (assert-channels-next-message-with-fn chans# #(= % msg#))))
+
+(let [get-token-hash (fn [kv-store token] (sd/token-data->token-hash (kv/fetch kv-store token)))
```

Comment:
We should create a new file for the tokens watch unit tests, e.g. `token_watch_test.clj`.